Learning_test_automation/Spotkanie8_Zadania/Hotel/WrapSeleniumMethod/WrappedSeleniumMethod.cs

```diff
@@ -0,0 +1,233 @@
+ using Locators.HomePage.BookRoomSection;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Support.UI;
+using FluentAssertions;
+using System.Collections.ObjectModel;
+using OpenQA.Selenium.Interactions;
+using System.Drawing;
+using Logger;
+using Microsoft.VisualBasic;
+
+namespace SeleniumDriver
+{
+    
+    public class WrappedSeleniumMethod : ISeleniumDriver
+    {
+        public readonly IWebDriver driver;
+        //public readonly NLogger nLogger;
+        NLogger nLogger = new NLogger();
+
+
+        public WrappedSeleniumMethod(IWebDriver webDriver)
+        {
+            driver = webDriver;
+            //NLogger nLogger = new NLogger();
+        }
+
+        public void OpenHomePageWebSite(string webUrlAdress)
+        {
+            var home = new HomePageBookRoomSection(driver);
+            home.driver.Navigate().GoToUrl(webUrlAdress);
+
+            try
+            {
+                nLogger.Info($"Opening the website: {webUrlAdress}");
+
+                string homeUrl = new WebDriverWait(driver, TimeSpan.FromSeconds(5))
+                                    .Until(_ => _.Url);
+
+                homeUrl.Should().Be(webUrlAdress);
+
+                nLogger.Info("Website opened successfully.");
+            }
+            catch (Exception ex)
+            {
+                nLogger.Error($"An error occurred while opening the website: {ex.Message}");
+                throw;
+            }
+            Console.WriteLine($"Current Directory: {Environment.CurrentDirectory}");
+        }
+
+        public void CountElementsOfGivenCollection(ReadOnlyCollection<IWebElement> collection, int amount)
```

Comment:
Tu nazwa sugeruje ze tylko liczymy elementy w kolekcji, a wewnatrz metody liczymy i sprawdzamy wzgledem oczekiwanej wartosci. Poprawilbym nazwe metody

Reply:
to będzie długa nazwa